lib/cloud_controller/paging/sequel_paginator.rb

```diff
@@ -74,7 +76,7 @@ def paginate_with_extension(dataset, per_page, page, table_name)
       end
 
       records = paged_dataset.all
-      records.each { |x| x.values.delete(:tmp_deferred_id) }
+      records.each { |x| x.values.delete(:tmp_deferred_id) } if records.first&.keys&.include?(:tmp_deferred_id)
```

Comment:
I prefer the `has_tmp_deferred_id` variable above if this is going to be an inline `if` statement